corehq/blobs/tests/test_export.py

```diff
@@ -78,5 +85,121 @@ def test_migrate_multimedia(self):
+
+    def tearDown(self):
+        for meta in self.blob_metas:
+            meta.delete()
+        self.db.close()
+
+    def test_33_x_1GB_blobs(self):
+        for __ in range(33):
+            meta = self.db.put(
+                MockBigBlobIO(zeros(), GB),
+                meta=new_meta(domain=self.domain_name, type_code=CODES.multimedia)
+            )
+            self.blob_metas.append(meta)
+
+        with NamedTemporaryFile() as out:
+            exporter = EXPORTERS['all_blobs'](self.domain_name)
+            exporter.migrate(out.name, force=True)
+
+            with tarfile.open(out.name, 'r:gz') as tgzfile:
+                self.assertEqual(
+                    set(tgzfile.getnames()),
+                    {m.key for m in self.blob_metas}
+                )
+
+    def test_1_x_33GB_blob(self):
+        meta = self.db.put(
+            MockBigBlobIO(zeros(), 33 * GB),
+            meta=new_meta(domain=self.domain_name, type_code=CODES.multimedia)
+        )
+        self.blob_metas.append(meta)
+
+        with NamedTemporaryFile() as out:
+            exporter = EXPORTERS['all_blobs'](self.domain_name)
+            exporter.migrate(out.name, force=True)
+
+            with tarfile.open(out.name, 'r:gz') as tgzfile:
+                self.assertEqual(
+                    set(tgzfile.getnames()),
+                    {m.key for m in self.blob_metas}
+                )
+
+
+class TestMockBigBlobIO(SimpleTestCase):
+
+    def test_zeros(self):
+        maybe_byte = next(zeros())
+        self.assertIsInstance(maybe_byte, bytes)
+
+    def test_read(self):
+        big_blob = MockBigBlobIO(zeros(), 100)
+        data = big_blob.read()
+        self.assertEqual(len(data), 100)
+
+    def test_read_size(self):
+        big_blob = MockBigBlobIO(zeros(), 100)
+        data = big_blob.read(50)
+        self.assertEqual(len(data), 50)
+
+    def test_read_chunks(self):
+        big_blob = MockBigBlobIO(zeros(), 100)
+        data1 = big_blob.read(25)
+        data2 = big_blob.read(100)
+        self.assertEqual(len(data1) + len(data2), 100)
+
+
+class MockBigBlobIO(RawIOBase):
+    """
+    A file-like object used for mocking a very large blob.
+
+    Consumes bytes from a generator up to a maximum size.
+    """
+    def __init__(self, bytes_gen, max_size):
+        self.generator = bytes_gen
+        self.size = max_size
+        self._consumed = 0
+
+    def readable(self):
+        return True
+
+    def seekable(self):
+        return False
+
+    def writable(self):
+        return False
+
+    def read(self, size=None):
+        size_remaining = self.size - self._consumed
+        if size is None or size < 0:
+            size = size_remaining
+        size = min(size, size_remaining)
+        if size > 0:
+            chunk = b''.join((next(self.generator) for __ in range(size)))
+            self._consumed = self._consumed + size
+            return bytes(chunk)
+
+    def readall(self):
+        return self.read(size=None)
```

Comment:
nit
```suggestion
        return self.read()
```

Reply:
lol, I was thinking "Explicit is better than implicit", but maybe I should have added, "... unless it's obvious".